langchain4j-pgvector/src/main/java/dev/langchain4j/store/embedding/pgvector/PgVectorEmbeddingStore.java

```
@@ -342,32 +441,137 @@ referenceVector, join(",", metadataHandler.columnsNames()), table, referenceVect
         return new EmbeddingSearchResult<>(result);
     }
 
+    private EmbeddingSearchResult<TextSegment> hybridSearch(EmbeddingSearchRequest request) {
+        Embedding referenceEmbedding = request.queryEmbedding();
+        String keywordQuery = request.query();
+
+        if (isNullOrBlank(keywordQuery)) {
+            return new EmbeddingSearchResult<>(List.of());
```

Maybe we should throw an exception instead? If user wants to do a hybrid search, but query is missing, something is wrong on their side, so we need to fail

---

Agree